### PR TITLE
Open session in view: false (users, import, studies) + latest dev ShUp

### PR DESCRIPTION
**Pull request by michaelkain**
- changes open-in-view to "false" for ms import, ms users, ms studies (not more)
- **MS Studies**:

1. StudyTags load them as eager with join (instead new select) for performance reasons, avoid N+1 effect
2. Better manage numberOfSubjects + Studies, that load today the entire sub-collection, and use a count
3. StudyLightDTO introduced: now used by getPublicStudiesData and getStudiesLight API endpoints
4. Use only one select to database to get numberOf-Subjects/-Exams for all studies, instead of N
5. Remove security overdoing from StudiesApi: on findStudies() + findStudiesLight(): why? 1) only studies with the correct rights are searched from the db in the repo already + 2) there is already a security postFilter on the service, that checks again on the rights; checking a third time the same thing and again going to the database is really security overdoing for performance costs.

- **front**:

1. getStudiesLight() API service call introduced and connected to home page
2. On home page: call to studies box (tested on qualif, with admin account), down from 2550ms to 600ms (in average), less than half of the time (tested: data are the same on the screen)

User type: **EXPERT** (as the one mostly editing study)
Local tests, done:
| Entity  | List | View | Create | Edit | Delete |
| ------------- | ------------- |------------- | ------------- | ------------- | ------------- |
| Study | OK | OK | OK | OK | OK |
| Subject | OK | OK | OK | OK | OK |
| Center | OK | OK | OK | OK | OK |
| Coil | OK | OK | OK | OK | OK |
| Acquisition equipment | OK | OK | OK | OK | OK |

| Import type  | Upload | Select series | Create sub+exam | Import | Download+viewer |
| ------------- | ------------- |------------- | ------------- | ------------- | ------------- |
| DICOM zip web| OK (start) | OK (upload) | OK (clinical context) | OK | OK |

User type: **ADMIN**
Local tests, done:
| Entity  | List | View | Create | Edit | Delete |
| ------------- | ------------- |------------- | ------------- | ------------- | ------------- |
| Study | OK | OK | OK | OK | OK |
| Subject | OK | OK | OK | OK | OK |
| Center | OK | OK | OK | OK | OK |
| Coil | OK | OK | OK | OK | OK |
| Acquisition equipment | OK | OK | OK | OK | OK |

| Import type  | Upload | Select series | Create sub+exam | Import | Download+viewer |
| ------------- | ------------- |------------- | ------------- | ------------- | ------------- |
| DICOM zip web| OK (start) | OK (upload) | OK (clinical context) | OK | OK |
| ShUp - Unit test zip import| OK (start) | OK (upload) | OK (clinical context) | OK | OK |
| ShUp - normal import| OK (start) | OK (upload) | OK (clinical context) | OK | OK |

### CODE REVIEW

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/center/service/CenterServiceImpl.java [hunk 1]
```diff
@@ -111,7 +111,9 @@ public Optional<Center> findById(final Long id) {
 	}
 	
 	public List<Center> findAll() {
-		return Utils.toList(centerRepository.findAll());
+		List<Center> centers = centerRepository.findAll();
+		centers.stream().forEach(c -> c.setAcquisitionEquipments(centerRepository.findDistinctAcquisitionEquipmentsByCenterId(c.getId())));
+		return Utils.toList(centers);
```
Comment (youennmerel) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/center/service/CenterServiceImpl.java:116]: Isn't centers already a List ?

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/dto/mapper/SubjectDecorator.java [hunk 2]
```diff
@@ -18,12 +18,18 @@ public class SubjectDecorator implements SubjectMapper {
...
+				return subjectToSubjectDTONoStudies(entity);
 			}
 		});
 		return new PageImpl<>(mappedPage);
 	}
 
+	@Override
+	public SubjectDTO subjectToSubjectDTONoStudies(Subject subject) {
+		final SubjectDTO subjectDTO = delegate.subjectToSubjectDTONoStudies(subject);
+		return subjectDTO;
```
Comment (youennmerel) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/dto/mapper/SubjectDecorator.java:30]: I think you can directly return `delegate.subjectToSubjectDTONoStudies(subject);` here

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java [hunk 3]
```diff
@@ -284,17 +283,14 @@ public boolean updateSubjectName(SubjectDTO subject) throws MicroServiceCommunic
...
 	@Override
 	public List<SimpleSubjectDTO> findAllSubjectsOfStudyId(final Long studyId) {
 		List<SimpleSubjectDTO> simpleSubjectDTOList = new ArrayList<>();
-		List<SubjectStudy> opt = subjectStudyRepository.findByStudy(studyRepository.findById(studyId).orElse(null));
-		if (opt != null) {
-			for (SubjectStudy rel : opt) {
+		List<SubjectStudy> studySubjects = subjectStudyRepository.findByStudyId(studyId);
+		Study studyWithTags = studyRepository.findStudyWithTagsById(studyId);
+		studySubjects.stream().forEach(ss -> ss.setStudy(studyWithTags));
+		if (studySubjects != null) {
```
Comment (youennmerel) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java:292]: Maybe return empty List after line 289 if studySubjects is null, to limit cyclomatic complexity ?

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java [hunk 4]
```diff
@@ -312,9 +308,11 @@ public List<SimpleSubjectDTO> findAllSubjectsOfStudyId(final Long studyId) {
 	@Override
 	public List<SimpleSubjectDTO> findAllSubjectsOfStudyAndPreclinical(final Long studyId, boolean preclinical) {
 		List<SimpleSubjectDTO> simpleSubjectDTOList = new ArrayList<>();
-		List<SubjectStudy> opt = subjectStudyRepository.findByStudy(studyRepository.findById(studyId).orElse(null));
-		if (opt != null) {
-			for (SubjectStudy rel : opt) {
+		List<SubjectStudy> studySubjects = subjectStudyRepository.findByStudyId(studyId);
+		Study studyWithTags = studyRepository.findStudyWithTagsById(studyId);
+		studySubjects.stream().forEach(ss -> ss.setStudy(studyWithTags));
+		if (studySubjects != null) {
```
Comment (youennmerel) [inline shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java:314]: Maybe return empty List after line 311 if studySubjects is null, to limit cyclomatic complexity ?

File: shanoir-ng-users/src/main/java/org/shanoir/ng/email/EmailServiceImpl.java [hunk 5]
```diff
@@ -205,9 +205,8 @@ public void notifyCreateAccountRequest(final User user, final String password) {
 			final String content = build("notifyCreateAccountRequest", variables);
 			messageHelper.setText(content, true);
 		};
-		LOG.error("User asked for an account" + user.getUsername());
+		LOG.info("User asked for an account: " + user.getUsername());
```
Comment (youennmerel) [inline shanoir-ng-users/src/main/java/org/shanoir/ng/email/EmailServiceImpl.java:208]: Better use `{}` here, instead of string concatenation

**Review by youennmerel — COMMENTED** [review]
Hi @michaelkain,

Some small remarks.

**Comment by youennmerel** [comment]
Hi @michaelkain,

Blocking bug while testing on OFSEp qualif.

GET /studies/coils fails with a 500 "Type definition error: [simple type, class org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor]"

This impact details of study, list of coils, details of study-card 

This type of logs are written in MS studies :
```
ofsep-qualif-studies-1  | 2024-03-06T16:54:29.600+01:00 ERROR 1 --- [nio-9902-exec-3] o.s.n.s.c.GlobalExceptionHandler         : Unexpected error in the rest service. 
ofsep-qualif-studies-1  | 
ofsep-qualif-studies-1  | org.springframework.http.converter.HttpMessageConversionException: Type definition error: [simple type, class org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor]
ofsep-qualif-studies-1  | 	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.writeInternal(AbstractJackson2HttpMessageConverter.java:489)
ofsep-qualif-studies-1  | 	at org.springframework.http.converter.AbstractGenericHttpMessageConverter.write(AbstractGenericHttpMessageConverter.java:114)
...
ofsep-qualif-studies-1  | Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: java.util.ArrayList[0]->org.shanoir.ng.coil.dto.CoilDTO["center"]->org.shanoir.ng.center.dto.CenterDTO["acquisitionEquipments"]->java.util.ArrayList[1]->org.shanoir.ng.acquisitionequipment.dto.AcquisitionEquipmentDTO["manufacturerModel"]->org.shanoir.ng.manufacturermodel.model.ManufacturerModel$HibernateProxy$ErSHx4no["hibernateLazyInitializer"])
ofsep-qualif-studies-1  | 	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
ofsep-qualif-studies-1  | 	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1308)
...
```

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/center/service/CenterServiceImpl.java (hunk 1 above)
Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/center/service/CenterServiceImpl.java:116]: done

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/dto/mapper/SubjectDecorator.java (hunk 2 above)
Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/dto/mapper/SubjectDecorator.java:30]: done

File: shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java (hunk 3 above)
Comment (michaelkain) [reply shanoir-ng-studies/src/main/java/org/shanoir/ng/subject/service/SubjectServiceImpl.java:292]: well seen, done

File: shanoir-ng-users/src/main/java/org/shanoir/ng/email/EmailServiceImpl.java (hunk 5 above)
Comment (michaelkain) [reply shanoir-ng-users/src/main/java/org/shanoir/ng/email/EmailServiceImpl.java:208]: done

**Comment by youennmerel** [comment]
> Hi @michaelkain,
> 
> Blocking bug while testing on OFSEP qualif.
> 
> GET /studies/coils fails with a HTTP 500 `Type definition error: [simple type, class org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor]`
> 
> This impact details of study, list of coils, details of study-card

Hi @michaelkain,

There's still an issue with details of a coil : `HTTP 500 : Could not write JSON: could not initialize proxy [org.shanoir.ng.manufacturermodel.model.ManufacturerModel#11] - no Session`

```
ofsep-qualif-studies-1  | 2024-03-15T11:20:05.153+01:00 ERROR 1 --- [nio-9902-exec-3] o.s.n.s.c.GlobalExceptionHandler         : Unexpected error in the rest service. 
ofsep-qualif-studies-1  | 
ofsep-qualif-studies-1  | org.springframework.http.converter.HttpMessageNotWritableException: Could not write JSON: could not initialize proxy [org.shanoir.ng.manufacturermodel.model.ManufacturerModel#11] - no Session
ofsep-qualif-studies-1  | 	at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.writeInternal(AbstractJackson2HttpMessageConverter.java:492)
...
ofsep-qualif-studies-1  | Caused by: com.fasterxml.jackson.databind.JsonMappingException: could not initialize proxy [org.shanoir.ng.manufacturermodel.model.ManufacturerModel#11] - no Session (through reference chain: org.shanoir.ng.coil.dto.CoilDTO["center"]->org.shanoir.ng.center.dto.CenterDTO["acquisitionEquipments"]->java.util.ArrayList[1]->org.shanoir.ng.acquisitionequipment.dto.AcquisitionEquipmentDTO["manufacturerModel"]->org.shanoir.ng.manufacturermodel.model.ManufacturerModel$HibernateProxy$5CzfAcTy["_links"])
ofsep-qualif-studies-1  | 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:402)
...
ofsep-qualif-studies-1  | Caused by: org.hibernate.LazyInitializationException: could not initialize proxy [org.shanoir.ng.manufacturermodel.model.ManufacturerModel#11] - no Session
ofsep-qualif-studies-1  | 	at org.hibernate.proxy.AbstractLazyInitializer.initialize(AbstractLazyInitializer.java:165)
ofsep-qualif-studies-1  | 	at org.hibernate.proxy.AbstractLazyInitializer.getImplementation(AbstractLazyInitializer.java:314)
...
```

Details of study, list of coils, details of study-card are now fine, though.

**Comment by michaelkain** [comment]
Hi @youennmerel, I have fixed the issue on the coil detail + an issue I found myself on the nbExams in the studies list. Happy further testing and with kind regards, Michael

**Comment by michaelkain** [comment]
Hi @youennmerel, I fixed one more issue on CT imports. Now it works. Happy further testing. I wait now for your next results. With kind regards, Michael

**Comment by youennmerel** [comment]
Hi @michaelkain,

Edit mode on coil fail with 
```
Error: Uncaught (in promise): TypeError: this.manufModels is undefined
9043/initEdit/<@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/main.5e0fb66301b81c9c.js:1:504767
invoke@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:7400
onInvoke@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/main.5e0fb66301b81c9c.js:1:101152
invoke@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:7340
run@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:2793
4924/</oe/<@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:17584
invokeTask@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:8018
onInvokeTask@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/main.5e0fb66301b81c9c.js:1:100968
invokeTask@https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/polyfills.5a71f46f3b9ace3b.js:1:7939
```

There also seems to be a problem with creation and listing of preclinical subjects :
```
Error: Uncaught (in promise): W1: {"headers":{"normalizedNames":{},"lazyUpdate":null},"status":500,"statusText":"OK","url":"https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/studies/subjects?clinical=false","ok":false,"name":"HttpErrorResponse","message":"Http failure response for https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/studies/subjects?clinical=false: 500 OK","error":{"code":500,"message":"failed to lazily initialize a collection of role: org.shanoir.ng.subject.model.Subject.subjectStudyList: could not initialize proxy - no Session","details":null}}
...

Cheers,
`

**Comment by michaelkain** [comment]
https://shanoir-qualif.irisa.fr/shanoir-ng/studies/studies/studyUser/260
[500] INTERNAL_SERVER_ERROR
https://shanoir-qualif.irisa.fr/shanoir-ng/studies/studies/studyUser/260
message : Could not write JSON: failed to lazily initialize a collection of role: org.shanoir.ng.study.model.StudyUser.centers: could not initialize proxy - no Session

**Comment by michaelkain** [comment]
500] INTERNAL_SERVER_ERROR
https://shanoir-qualif.irisa.fr/shanoir-ng/studies/studies/259
message : A collection with cascade="all-delete-orphan" was no longer referenced by the owning entity instance: org.shanoir.ng.tag.model.Tag.subjectStudyTags

**Comment by michaelkain** [comment]
2 more errors remain:
- ask access to public study in studies list (locked study)
- no zip files inside challenges bubble

**Comment by michaelkain** [comment]
5 test cases: profile OFSEP, adapt dev profile for OFSEP
1) new subject not existing with hash in entire database
test: OK, new subject created, new exam
2) subject exists with hash in study-no-access
test: OK, new subject created, new exam
3) subject existing with hash in another accessible study
test: OK, existing subject used, new exam created in other study
4) subject existing with hash in importing study, new exam
test: OK, existing subject used, new exam created in same study
5) subject existing with hash in importing study, existing exam
test: OK, existing subject used, existing exam used, double amount of series

**Comment by michaelkain** [comment]
5 test cases: profile Neurinfo, user decides
1) new subject not existing with hash in entire database
test: OK, new subject created, new exam
2) subject exists with hash in study-no-access test: OK, new subject created, new exam
test: OK, new subject created, new exam
3) subject existing with hash in another accessible study test: OK, existing subject used, new exam created in other study
test: OK, as user wishes
4) subject existing with hash in importing study, new exam test: OK, existing subject used, new exam created in same study
same test case for Neurinfo
test: OK
5) subject existing with hash in importing study, existing exam test: OK, existing subject used, existing exam used, double amount of series
test: OK, even tested with multiple-exams

**Comment by michaelkain** [comment]
- test preclinical import and display
	- all 5 entity tables are loaded for user type USER
		- test: OK
	- test bruker import, demo_appning on your branch
		- test: OK, but
		- strange, no study card selection, new issue created https://github.com/fli-iam/shanoir-ng/issues/2346
	- display preclinical subjects + exams and open new created entities
		- test: OK now (bug found and solved)

**Review by youennmerel — CHANGES_REQUESTED** [review]
Hi @michaelkain,
Tested on OFSEP qualif, as admin, some problems :
- `TypeError: e.studyCenterList is null` when creating new center
   - not reproduced on develop (neurinfo qual) 
- preclinical subject list is empty, despite 1 existing preclinical subjects
- 500 error when displaying PREEXISTING preclinical subject (https://shanoir-ofsep-qualif.irisa.fr/shanoir-ng/preclinical-subject/details/5385)
  - creation of a new subject through import is OK, it's listed and can be displayed

**Review by youennmerel — APPROVED** [review]

**Pull request merged**